test/util/malicious/hasher.go

```diff
@@ -80,9 +101,19 @@ func (n *Hasher) Write(data []byte) (int, error) {
 	switch ln {
 	// inner nodes are made up of the nmt hashes of the left and right children
 	case n.Size() * 2:
+		// check the format of the data
+		leftChild := data[:n.Size()]
+		rightChild := data[n.Size():]
+		if err := n.ValidateNodes(leftChild, rightChild); err != nil {
+			return 0, err
+		}
 		n.tp = NodePrefix
 	// leaf nodes contain the namespace length and a share
 	default:
+		// validate the format of the leaf
+		if err := n.ValidateLeaf(data); err != nil {
+			return 0, err
+		}
```

Comment:
see above comment 

> I tried to keep the changes as minimal as possible. This means that instead of remove all the validation entirely, we're now only removing the pieces specific to ordering the namespaces